src/main/java/com/example/springbootapp/exception/EmployeeExceptionHandler.java

```diff
@@ -33,6 +39,37 @@ public ResponseError employeeNotSaved(EmployeeNotSavedException exception) {
         return new ResponseError(HttpStatus.BAD_REQUEST, exception.getMessage());
     }
 
+    @ResponseBody
+    @ExceptionHandler(ConstraintViolationException.class)
+    @ResponseStatus(HttpStatus.BAD_REQUEST)
+    public ValidationResponseError onConstraintValidationException(
+            ConstraintViolationException exception
+    ) {
+        LOGGER.error(exception.getMessage(), exception);
+        final List<Violation> violations = exception.getConstraintViolations().stream()
+                .map(
+                        violation -> new Violation(
+                                violation.getPropertyPath().toString(),
+                                violation.getMessage()
+                        )
+                )
+                .collect(Collectors.toList());
+        return new ValidationResponseError(HttpStatus.BAD_REQUEST,violations);
```

Comment:
We discussed that we need the same object for all error responses. Why did you decide to use the other object?

Reply:
Please add the optional List<Violation> field in ResponseError and use it when you have the list of such violations

Reply:
optional, do you mean java.util.Optional?